### PR TITLE
Verify customer screen 

### PR DESCRIPTION
**Pull request by imcvampire**

### CODE REVIEW

File: src/components/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default as BottomContainer } from './BottomContainer'
```
Comment (jsw324) [inline src/components/index.js:1]: let's remove this file as it'll grow out of control. We can import the components we need directly from the component file

File: src/components/BottomContainer/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { styled } from '@mui/material/styles'
+
+const BottomContainer = styled('div')({
```
Comment (jsw324) [inline src/components/BottomContainer/index.js:3]: this maybe belongs in an 'atoms' folder within components folder so we can have one file for all the little styled components we create.  Not a deal breaker just a thought.  The intent of the components folder is to have a repository of all our reusable components -- something to think about is how easy it'll be to find items and reason about as it grows.

File: src/lib/apollo/index.js [hunk 3]
```diff
@@ -19,12 +19,12 @@ const defaultOptions = {
 
 const withApolloClient = App => {
   const WithApolloClient = props => {
-    const httpLink = createHttpLink({
+    const uploadLink = createUploadLink({
```
Comment (jsw324) [inline src/lib/apollo/index.js:22]: 😎

File: src/pages/VerifyConsumer/index.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  Backdrop,
+  CircularProgress
+} from '@mui/material'
+import { styled } from '@mui/material/styles'
+import { useMutation } from '@apollo/client'
+
+import CloudUploadRounded from '@mui/icons-material/CloudUploadRounded'
+
+import { VERIFY_CONSUMER } from './mutation'
+import { BottomContainer } from '../../components'
```
Comment (jsw324) [inline src/pages/VerifyConsumer/index.js:18]:
```suggestion
import BottomContainer from '../../components/BottomContainer'
```

File: src/pages/VerifyConsumer/index.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      const file = event.target.files[0]
+      setFile(file)
+      setImage(URL.createObjectURL(file))
+    }
+  }
+
+  const handleNextStep = async () => {
+    if (!image) return
+
+    if (currentPage === INFO_PAGE) {
```
Comment (jsw324) [inline src/pages/VerifyConsumer/index.js:58]: this is confusing. I think we can have a better flow to these components.

We want to avoid nested `if` statements as they're a source of confusion for other developers

File: src/pages/VerifyConsumer/index.js [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+import {
+  Button,
+  Container,
+  Typography,
+  Box,
+  Grid,
+  TextField,
+  Backdrop,
+  CircularProgress
+} from '@mui/material'
```
Comment (jsw324) [inline src/pages/VerifyConsumer/index.js:11]: lets import each of these components individually

`import Button from '@mui/material/Button`

Reason being we'll be importing the entire library by doing it that way, especially in development: https://mui.com/guides/minimizing-bundle-size/#development-environment

File: src/pages/VerifyConsumer/index.js [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+              onChange={(e) => setEmail(e.target.value)}
+            />
+            <Box mt={2} mb={1}>
+              <Typography variant='title'>Twitter</Typography>
+            </Box>
+            <TextField fullWidth placeholder='Enter your twitter id' />
+          </Box>
+          )
+        : (
+          <Box mt={1}>
```
Comment (jsw324) [inline src/pages/VerifyConsumer/index.js:102]: this should be it's own component.  If you feel we can reuse it let's move it to `components/photo_uploader`

File: src/pages/VerifyConsumer/index.js [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    } else {
+      setCurrentPage(INFO_PAGE)
+    }
+  }
+
+  return (
+    <Container style={{ marginTop: '140px' }}>
+      {currentPage === INFO_PAGE
+        ? (
+          <Box>
```
Comment (jsw324) [inline src/pages/VerifyConsumer/index.js:81]: this should be it's own component. We can move it to a file in this directory.

**Review by jsw324 — CHANGES_REQUESTED** [review]
A few questions:

- Do we need/want a way to handle users that are already logged in?
- Do we want this to happen at the root `/` URL or should it occur at a different path?

Please feel free to take liberties with the design, don't feel like you need to adhere to that flow I sent.

File: src/pages/VerifyConsumer/index.js (hunk 5 above)
Comment (imcvampire) [reply src/pages/VerifyConsumer/index.js:58]: What do you think if we redirect to a new route?

File: src/pages/VerifyConsumer/index.js (hunk 6 above)
Comment (imcvampire) [reply src/pages/VerifyConsumer/index.js:11]: What do you think if we use the babel plugin for that? https://mui.com/guides/minimizing-bundle-size/#option-2

**Comment by imcvampire** [comment]
> A few questions:
> 
> * Do we need/want a way to handle users that are already logged in?
> * Do we want this to happen at the root `/` URL or should it occur at a different path?
> 
> Please feel free to take liberties with the design, don't feel like you need to adhere to that flow I sent.

1. IIRC, we don't need to remember the authentication. But if we need it, we can store it securely on the cookie. 
2. If there is no other screen (e.g., Introduction, ...), I think we can use `/`.

File: src/pages/VerifyConsumer/index.js (hunk 5 above)
Comment (jsw324) [reply src/pages/VerifyConsumer/index.js:58]: i think that could work - the most important thing is to avoid confusing logic with many 'if' clauses.  We can do that through separating the components or a new route.

I'm not sure we're getting any benefits to a new route as we're never going to send a user directly to step 2.

Maybe [this component ](https://mui.com/components/steppers/) could work?

File: src/components/BottomContainer/index.js (hunk 2 above)
Comment (imcvampire) [reply src/components/BottomContainer/index.js:3]: Personally, I flat all components but it's hard to classify client-parent components and I don't want to re-order if a new use case is introduced. 

But I think it's great if we follow your suggestion and see how things work.

File: src/pages/VerifyConsumer/index.js [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+            <Box mb={1}>
+              <Typography variant='title'>Name</Typography>
+            </Box>
+            <TextField fullWidth placeholder='Enter your name' />
+            <Box mt={2} mb={1}>
+              <Typography variant='title'>Email</Typography>
+            </Box>
+            <TextField
```
Comment (jsw324) [inline src/pages/VerifyConsumer/index.js:89]: can we validate email is correct? In other projects we use [react-hook-forms](https://react-hook-form.com/) to help clean up forms and validation

File: src/components/BottomContainer/index.js (hunk 2 above)
Comment (jsw324) [reply src/components/BottomContainer/index.js:3]: why would you have to reorder?

File: src/pages/VerifyConsumer/index.js (hunk 6 above)
Comment (jsw324) [reply src/pages/VerifyConsumer/index.js:11]: No reason to introduce more complexity, let's just import each component individually.

File: src/components/BottomContainer/index.js (hunk 2 above)
Comment (imcvampire) [reply src/components/BottomContainer/index.js:3]: Ah, no. If `atom`s are styled components, I believe we don't need to re-order if they could be changed from a child to a parent component.

File: src/pages/VerifyConsumer/index.js (hunk 5 above)
Comment (imcvampire) [reply src/pages/VerifyConsumer/index.js:58]: I think we're going to send a user to step 2. After selecting a picture and confirming it (step 1), the user will be redirected to step 2 for other inputs. Moreover, having separate pages could improve readability and/or editability if each step is large enough. 

For your suggested component, I think it's suitable for small steps. In our use case, we have 5 steps in total, and each page is different. From the implementation aspect, we might have 5 render functions, and a check (`map`, `if`, `switch`, ...) to use the correct function based on the index. Seems like not an easy task. 

 If we use separate routes, the number of LoC and the complexity will be smaller, which is easier to maintain.

Comment (jsw324) [reply src/pages/VerifyConsumer/index.js]: i meant we're never going to send a user directly to step 2 

i'm good with using separate routes, that works.

**Comment by jsw324** [comment]
This is better...i would like to recognize that i'm logged in, if possible as it's inconvenient to make a user take a new selfie each time he/she logs in to see their photos.

I'm also thinking this can be a single step - it's only 3 fields in the form and we're not verifying the upload was successfully before moving to step 2 so i don't see why this shouldn't be one screen.  I'll approve so we're not blocked but i think this would be better in a single screen.

Thoughts?

**Comment by imcvampire** [comment]
> This is better...i would like to recognize that i'm logged in, if possible as it's inconvenient to make a user take a new selfie each time he/she logs in to see their photos.
> 
> I'm also thinking this can be a single step - it's only 3 fields in the form and we're not verifying the upload was successfully before moving to step 2 so i don't see why this shouldn't be one screen. I'll approve so we're not blocked but i think this would be better in a single screen.
> 
> Thoughts?

I've thought about that after some time, I think your design makes sense so I stick with it. 

> Please test this before merging as i'm getting a graphQL error as well

Yes, seems like it's an edge case. I'm checking it with Hong. 

BTW, seems like I don't have permission to merge.

**Comment by jsw324** [comment]
@imcvampire not sure why you can't merge - i merged for you and i'll check the settings.  

< i think your design makes sense so I stick with it.

 Are you saying you're going to move it to one screen with the image uploader and the 3 inputs? I do think that's better but we can always do it later :)
